### ISSUE TITLE
Generate: have an example on each `LogitsProcessor` class docstring

### ISSUE THREAD

**Issue by gante** [huggingface/transformers#24783]
# Context

`.generate()` can be extensively manipulated through `LogitsProcessor` (and `LogitsWarper`) classes. Those classes are the code implementation behind flags like `temperature` or `top_k`. 

Most of our `LogitsProcessor` classes have a docstring that briefly describes their effect. However, unless you are an expert in text generation, it's hard to fully grasp the impact of using each class. In some cases, it is also non-trivial to prepare the arguments to initialize the `LogitsProcessor` class. As such, each class should have a clear usage example with `.generate()` in their docstring! 💪 

Here is an example: [SequenceBiasLogitsProcessor docstring](https://github.com/huggingface/transformers/blob/f1732e1374a082bf8e43bd0e4aa8a2da21a32a21/src/transformers/generation/logits_process.py#L559). Contrarily to the other classes (at the time of writing), we can quickly learn how to use it just by reading its docstring. We are also immediately aware of a few caveats 🤓 

Bonus points: our docstring examples are part of our CI, so we would be beefing up our tests to ensure we don't add regressions 🤗 

This issue is part of the [text generation docs rework](https://github.com/huggingface/transformers/issues/24575).

# How to participate?
1. Ensure you've read our contributing [guidelines](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md) 📜 
2. Claim your `LogitProcessor` class in this thread (confirm no one is working on it). You can check the full list of classes below, and you can find their implementation in [this file](https://github.com/huggingface/transformers/blob/main/src/transformers/generation/logits_process.py) 🎯 
    - You may need to do some detective work to fully understand the purpose of the class. For instance, some classes were created as part of a paper to be applied to any model, others are model-specific, and some exist to avoid weird bugs 🕵️ 
    - Looking at the git history is a great way to understand how a `LogitsProcessor` came to be.
3. Implement your changes, taking the [SequenceBiasLogitsProcessor docstring](https://github.com/huggingface/transformers/blob/f1732e1374a082bf8e43bd0e4aa8a2da21a32a21/src/transformers/generation/logits_process.py#L559) as reference  💪 
    - Add a clear example that calls the processor through `.generate()`. Make sure the example's outputs are correct and that the model used in the test is a small model (anything larger than GPT2 needs explicit approval);
    - If you feel like the original docstring could be better, feel free to enhance it as well!
    - Don't forget to run `make fixup` before your final commit.
4. Open the PR and tag me in it  🎊 

# Tracker
- [x] MinNewTokensLengthLogitsProcessor
- [x] TemperatureLogitsWarper
- [x] RepetitionPenaltyLogitsProcessor
- [ ] EncoderRepetitionPenaltyLogitsProcessor
- [x] TopPLogitsWarper
- [ ] TopKLogitsWarper
- [ ] TypicalLogitsWarper
- [x] EpsilonLogitsWarper
- [x] EtaLogitsWarper
- [x] NoRepeatNGramLogitsProcessor
- [ ] EncoderNoRepeatNGramLogitsProcessor
- [x] SequenceBiasLogitsProcessor
- [x] NoBadWordsLogitsProcessor
- [ ] PrefixConstrainedLogitsProcessor
- [x] HammingDiversityLogitsProcessor
- [ ] ForcedBOSTokenLogitsProcessor
- [ ] ForcedEOSTokenLogitsProcessor
- [ ] InfNanRemoveLogitsProcessor
- [ ] ExponentialDecayLengthPenalty
- [ ] LogitNormalization
- [ ] SuppressTokensAtBeginLogitsProcessor
- [ ] SuppressTokensLogitsProcessor
- [ ] ForceTokensLogitsProcessor
- [ ] WhisperTimeStampLogitsProcessor
- [ ] ClassifierFreeGuidanceLogitsProcessor

**Comment by nablabits** [comment]
Hi @gante  I'm happy to give one of these a go as it seems a nice learning experience... `TemperatureLogitsWarper` feels as good as anyone else, can you assign it to me if it's not taken yet?

**Comment by gante** [comment]
@nablabits thank you for your interest. It's all yours! 

(The assignment is informal -- the first one to mention a certain class gets automatically assigned to it 🤗 )

**Comment by SoyGema** [comment]
Hello @gante 👋
Thanks for opening up this contribs .
Would like to give it a try to [NoBadWordsLogitsProcessor ](https://github.com/huggingface/transformers/blob/f1732e1374a082bf8e43bd0e4aa8a2da21a32a21/src/transformers/generation/logits_process.py#L725).  Alredy tried `bad_words_ids` argument and looking forward to digging and learning more into the impact of `eos_token_id` for this class. 

LKM if that works for you !

**Comment by Rishab26** [comment]
Hey @gante 👋 

I would like to try [RepetitionPenaltyLogitsProcessor](https://github.com/huggingface/transformers/blob/5bb4430edc7df9f9950d412d98bbe505cc4d328b/src/transformers/generation/logits_process.py#L194)

I hope that works!

**Comment by Rishab26** [comment]
Hey Shauray (@shauray8), Nice one! I can see you have opened a PR for the RepetitionPenaltyLogitsProcessor. I was working on it.  Just a note for next time, please go through the "How to Participate" and confirm no one is working on it.

@gante I'll look at other Processor Classes soon and take up something else. 👍

**Comment by shauray8** [comment]
Hey @Rishab26, I didn't go through the comments and I appreciate your understanding.

**Comment by SoyGema** [comment]
Hey @Rishab26 I would like to highlight your level of empathy towards this OSS Governance issue.
IMO, example of level 4  level in [trust-level system](https://blog.discourse.org/2018/06/understanding-discourse-trust-levels/) HF relies upon . 

🤗 Thanks for setting a positive example for the Open Source Community. 🤗

**Comment by SoyGema** [comment]
Hey Folks!
Still working on this, having fun though.
Im opening the WIP in this [repo](https://github.com/SoyGema/contrib_schema/)  in case someone wants to have a look before PR.
Have already some things , but giving it a careful thought to the example and digging into some things.
Found this gem #22168 !

**Comment by mobley-trent** [comment]
Hey @gante I would like to work on `SuppressTokensLogitsProcessor` 🙂

**Comment by Rishab26** [comment]
Hey @gante I am working on `NoRepeatNGramLogitsProcessor` 🙂

**Comment by akshayamadhuri** [comment]
hey @gante I am working on  **TypicalLogitsWarper**

**Comment by ashishthomaschempolil** [comment]
hey @gante I would like to work on **EtaLogitsWarper** 😊

**Comment by chiral-carbon** [comment]
Hi! May I claim `TopPLogitsWrapper`?

**Comment by TheTensorDude** [comment]
Hi @gante, I would like to work on `LogitNormalization`.

**Comment by nablabits** [comment]
Hey @gante, yes, me again :upside_down_face: , are you happy for me to pick `MinNewTokensLengthLogitsProcessor`?

**Comment by MFYLM** [comment]
Hi @gante can I try to work on `LogitNormalization`?

**Comment by shipra101** [comment]
Hi @gante , can I try working on `ForceTokensLogitsProcessor`?

**Comment by adtygan** [comment]
Hello @gante, I would like to work on `InfNanRemoveLogitsProcessor`. I see no one has referred to it in the comments section. But just wanted to make sure no one is working on it. Could you please confirm?

**Comment by gante** [comment]
> hi @gante i am working on EncoderRepetitionPenaltyLogitsProcessor, is there a way I can create a new issue named EncoderRepetitionPenaltyLogitsProcessor l then link that here and it shows active in the list?
like this, you can put a checklist there

We can, but I don't see the benefit of it, so I'm not convinced about having this extra step :D

**Comment by jessthebp** [comment]
Hi @gante , I'd love to try writing examples for `HammingDiversityLogitsProcessor` !

**Comment by sanjeevk-os** [comment]
Hi @gante , 
Working on [EpsilonLogitsWarper](https://github.com/huggingface/transformers/blob/a6e6b1c622d8d08e2510a82cb6266d7b654f1cbf/src/transformers/generation/logits_process.py#L442)

**Comment by ajanitshimanga** [comment]
Hi @gante ,
Can I work on TopPLogitsWarper? Thanks! :)

**Comment by chiral-carbon** [comment]
> Hi @gante , Can I work on TopPLogitsWarper? Thanks! :)

Hi! I'm already working on it.

**Comment by ajanitshimanga** [comment]
> > Hi @gante , Can I work on TopPLogitsWarper? Thanks! :)
> 
> Hi! I'm already working on it.

Ah I see, It looks like you spelled it as `TopPLogitsWrapper` rather than `TopPLogitsWarper` so I couldn't find it with a CTRL + F. I'll find a different one. 👍

**Comment by chiral-carbon** [comment]
> > > Hi @gante , Can I work on TopPLogitsWarper? Thanks! :)
> > 
> > 
> > Hi! I'm already working on it.
> 
> Ah I see, It looks like you spelled it as `TopPLogitsWrapper` rather than `TopPLogitsWarper` so I couldn't find it with a CTRL + F my bad! I'll take up SuppressTokensAtBeginLogitsProcessor, don't want to steal that from you haha 👍

My bad, I'll correct it. Thanks!

**Comment by jprivera44** [comment]
Hello, I'll work on the WhisperTimeStampLogitsProcessor.

**Comment by pokjay** [comment]
@gante While working on the docs, I noticed there might be an issue with the current implementation of ExponentialDecayLengthPenalty.

The processor is intended to exponentially increase the score of the eos_token_id after start_index has been reached, allowing generating shorter sequences without having a hard cutoff.

When working with shorter sequences (Up to ~200) it doesn't necessarily cut the sequence, no matter how large the decay factor is.
In the following line, the processor attempts to increase the score of EOS. However when EOS score is negative, this actually decreases the score, as the exponent will be positive. As I understand, giving a negative decay factor won't work as well due to the power. Due to this it will only succeed if EOS becomes positive.
https://github.com/huggingface/transformers/blob/f1732e1374a082bf8e43bd0e4aa8a2da21a32a21/src/transformers/generation/logits_process.py#L982

Two questions:

1. Is this actually an issue?
2. If it is, I believe I can fix it. Should I open an issue, or just fix it with the docs PR?

**Comment by pranith7** [comment]
Hey @gante I am working on **ClassifierFreeGuidanceLogitsProcessor**

**Comment by martinbrose** [comment]
Hi @gante, I'd like to work on the `EncoderNoRepeatNGramLogitsProcessor`.

**Comment by martinbrose** [comment]
> Hey @gante I am working on `NoRepeatNGramLogitsProcessor` 🙂

Hi @Rishab26, just quickly stating that you did do the work on `EncoderNoRepeatNGramLogitsProcessor`. It doesn't show as taken in the tracker above. So I was surprised it was already done... 

Hope this comment now updates the tracker.

**Comment by gante** [comment]
Hey everyone! `.generate()` documentation is one a focus of the next release, so I'm going to fast-track this issue internally. Thank you for your participation -- your contributions help our ecosystem thrive 💛 

👉 What does this mean?
1. I will NOT be accepting new assignments in this thread (and, as such, I'm closing it)
2. I will be accepting existing and new PRs for assigned logits processors over the next week. After that, I will take the remaining classes 🤗
